nsq/protocol.py

```diff
@@ -61,12 +61,12 @@ def unpack_response(data):
     return frame, data[4:]
 
 
-def decode_message(data):
+def decode_message(data, timeout=None):
```

Comment:
I don't think setting the timeout here is ideal, `decode_message` isn't very idiomatic when we also use it to pass the message `timeout`. The best alternative I can think of is to keep track of timeouts by default, but we'd see the additional overhead you mentioned.